src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -203,40 +201,47 @@ public void unselectDeck() {
 
     @Override
     public Optional<Deck> getSelectedDeck() {
-        return selectedDeck;
+        return Optional.ofNullable(selectedDeck);
     }
 
     @Override
     public String getSelectedDeckName() {
-        return selectedDeck.get().getDeckName();
+        return Optional.ofNullable(selectedDeck)
+                .map(Deck::getDeckName)
+                .orElse("None");
     }
 
+    /* Review Operations */
+
     /**
      * Starts a new review session based on deckIndex selected
-     * @param deckIndex
+     * @param deckIndex Index of the deck
      */
     public void reviewDeck(Index deckIndex) {
         int zeroBasesIdx = deckIndex.getZeroBased();
         Deck deckToReview = filteredDecks.get(zeroBasesIdx);
         List<Card> cardList = new FilteredList<>(
                 masterDeck.getCardList(), new CardInDeckPredicate(deckToReview)
         );
-        currReview = Optional.of(new Review(deckToReview, cardList));
-    };
+        currReview = new Review(deckToReview, cardList);
+    }
 
     @Override
-    public Review getReview() {
-        return currReview.orElse(null);
+    public Optional<Review> getReview() {
+        return Optional.ofNullable(currReview);
     };
 
     @Override
     public void endReview() {
-        currReview = Optional.empty();
+        currReview = null;
```

Comment:
I think the predicate shouldn't be set becoz we can start a review at any time (whether a Deck is selected or not) and so when we end the review it should return to the previous state. Ending the review does not change the selectedDeck state so the filteredCards should not be changed either.